src/TraceEvent/Parsers/ClrTraceEventParser.cs

```diff
@@ -5727,7 +5727,7 @@ private long[] GetIntPtrArray(int offset, int count)
             return arr;
         }
 
-        private const int maxGenData = (int)Gens.Gen0After + 1;
+        private const int maxGenData = (int)Gens.GenCount;
```

Comment:
I need to double-check this code path, something doesn't sound right here that we are unconditionally setting maxGenData to 5 but then use that to compute offset for parsing trace data.

Reply:
Turn out it wasn't wrong. `Gen0After` was 4, and therefore the new `GenCount` is the old `Gen0After + 1`.
To clarify, I just hardcoded it to 5. This number is used only for parsing `GCHeapHistory` prior to version 3, so it has to be 5.

Reply:
it actually does have something to do with max gen count, as in it was (max gen count + 1) since we had an extra one for gen0. could you please add a comment there to explain why this is 5? and make sure to mention that this is only used to parse older traces (as the newer versions no longer use it anymore.